### PR TITLE
mgr/cephadm: fix tuned profiles getting removed if name has dashes

### PR DESCRIPTION
**Pull request by adk3798**
Previously, due to the way it was just splitting the file name on
dashes to try and get the profile name, any profile name with dashes
was not getting properly matched and would therefore get marked
stray and removed. This new strategy instead tries to match the actual
expected file name.

Fixes: https://tracker.ceph.com/issues/57405

Signed-off-by: Adam King <adking@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [x] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by adk3798** [comment]
jenkins retest this please

File: src/pybind/mgr/cephadm/tuned_profiles.py [hunk 1]
```diff
@@ -47,16 +47,39 @@ def _write_all_tuned_profiles(self) -> None:
...
+        SHOULD be on the host. Then if we see any file names that don't match this, but
+        DO include "-cephadm-tuned-profile.conf" (implying they're from us), remove them.
+        """
         if host in self.mgr.offline_hosts:
             return
         cmd = ['ls', SYSCTL_DIR]
         found_files = self.mgr.ssh.check_execute_command(host, cmd).split('\n')
         found_files = [s.strip() for s in found_files]
+        profile_names = list(set([pname for pnames in [list(profile.keys())
+                             for profile in profiles] for pname in pnames]))
```
Comment (rkachach) [inline src/pybind/mgr/cephadm/tuned_profiles.py:76]:
```suggestion
        profile_names = sum( [[*p] for p in profiles], [])  # extract all profiles names
        profile_names = list(set(profile_names))  # keep unique names
```

**Review by rkachach — APPROVED** [review]

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2022-09-12_11:00:28-orch:cephadm-wip-adk-testing-2022-09-11-1805-distro-default-smithi/

failures tracked by https://tracker.ceph.com/issues/57287 and unrelated to any PRs in this run

File: src/pybind/mgr/cephadm/tuned_profiles.py (hunk 1 above)
Comment (adk3798) [reply src/pybind/mgr/cephadm/tuned_profiles.py]: `sum( [[*p] for p in profiles], [])` is nice. Didn't know about that format for getting all the keys in list of dicts

**Comment by adk3798** [comment]
jenkins test make check

**Pull request merged**